libcloud/storage/drivers/s3.py

```diff
@@ -1029,7 +1056,9 @@ def __init__(self, key, secret=None, secure=True, host=None, port=None,
         self.name = 'OUTSCALE Ceph RGW S3 (%s)' % (region)
         self.ex_location_name = region
         self.region_name = region
-        self.connectionCls = S3RGWOutscaleConnection
+        signature_version = kwargs.get('signature_version', 'aws2')
+        self.connectionCls =\
```

Comment:
Since `connectionCls` is a class-level attribute (and not an instance one), I believe this won't work correctly if multiple instances of this class are instantiated and each one uses a different signature version.

For example:

```python
cls1 = S3RGWOutscaleStorageDriver(..., signature_version='aws2')
cls2 = S3RGWOutscaleStorageDriver(..., signature_version='aws4')
```

The version of the class which was instantiated last wins so connectionCls of cls1 would also be set to `S3RGWOutscaleConnectionAWS4` after the second class is instantiated.

Another option, which I believe should work correctly, is to utilize `ex_connection_class_kwargs` method and pass correct value for the `version` argument to the `connectionCls` depending on the requested signature version.